src/DataTypes/FieldToDataType.cpp

```diff
@@ -136,8 +136,39 @@ DataTypePtr FieldToDataType<on_error>::operator() (const Array & x) const
     DataTypes element_types;
     element_types.reserve(x.size());
 
+    auto checkIfConversionSigned = [&](bool& has_signed_int, bool& has_uint64, bool& uint64_could_opposite,
```

Comment:
I see we have 2 identical lambdas here and during Map processing. Let's move this code to a separate function in anonimous namespace in this cpp file.

Reply:
And also the logic is the same here and later during Map processing. Maybe we can extract all this logic to a separate function?
Like
`void convertUInt64ToInt64IfPossible(DataTypes & data_types)`

So the code here will look like this:
```
...
for (const Field & elem : x)
        element_types.emplace_back(applyVisitor(*this, elem));
convertUInt64ToInt64IfPossible(element_types);
...
```

It will be much better for future readers

Reply:
OK, I'll handle it.